src/components/Widget.tsx

```diff
@@ -94,17 +88,14 @@ export const DialogWrapper = styled.div`
   }
 `
 
-export interface WidgetProps extends TransactionEventHandlers {
+export interface WidgetProps extends TransactionEventHandlers, Web3Props, WidgetEventHandlers {
   theme?: Theme
   locale?: SupportedLocale
-  provider?: Eip1193Provider | JsonRpcProvider | null
```

Comment:
These are more complex types, so they are used by extending `Web3Props`, which is less error-prone because it is less duplication.